src/GetDataCapableTrait.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Dhii\Data\Object;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * Functionality for data retrieval.
+ *
+ * @since [*next-version*]
+ */
+trait GetDataCapableTrait
+{
+    /**
+     * Retrieve data, all or by key.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null $key The key, for which to get the data,
+     *                                    or null to get all data as an array.
+     *
+     * @return mixed|array The value for the specified key, or all data as an key-value map.
+     */
+    protected function _getData($key = null)
+    {
+        if (!is_null($key) && !is_string($key) && !($key instanceof Stringable)) {
+            throw $this->_createInvalidArgumentException($this->__('Data key must be stringable'), null, null, $key);
+        }
+
+        $store = $this->_getDataStore();
+
+        // Return whole set
+        if (is_null($key)) {
+            return (array) $store;
+        }
+
+        $key = (string) $key;
```

Comment:
Can we abstract the normalization of the string here to allow implementations to use the [`NormalizeStringCapableTrait`](https://github.com/Dhii/normalization-helper-base/blob/task/initial-classes/src/NormalizeStringCapableTrait.php)? If so, the same thing applies for the other traits in this package that normalize a key.